src/Testcontainers.Pulsar/PulsarBuilder.cs

```diff
@@ -0,0 +1,95 @@
+
+namespace Testcontainers.Pulsar
+{
+  /// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+  [PublicAPI]
+  public sealed class PulsarBuilder : ContainerBuilder<PulsarBuilder, PulsarContainer, PulsarConfiguration>
+  {
+    public const string PulsarImage = "apachepulsar/pulsar-all:2.11.0";
+
+    public ushort PulsarPort = 6650;
+    public ushort PulsarAdminPort = 8080;
+    public ushort PulsarSQLPort = 8081;
+
+    public const string StartupScriptFilePath = "/testcontainers.sh";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="KafkaBuilder" /> class.
+    /// </summary>
+    public PulsarBuilder()
+        : this(new PulsarConfiguration())
+    {
+      DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="KafkaBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    private PulsarBuilder(PulsarConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+      DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override PulsarConfiguration DockerResourceConfiguration { get; }
+
+    /// <inheritdoc />
+    public override PulsarContainer Build()
+    {
+      this.Validate();
+      return new PulsarContainer(this.DockerResourceConfiguration, TestcontainersSettings.Logger);
+    }
+    /// <inheritdoc />
+    protected override PulsarBuilder Init()
+    {
+      return base.Init()
+          .WithImage(PulsarImage)
+          .WithPortBinding(this.PulsarPort, this.PulsarPort)
+          .WithPortBinding(this.PulsarAdminPort, this.PulsarAdminPort)
+          .WithPortBinding(this.PulsarSQLPort, this.PulsarSQLPort)
+          .WithEnvironment("PULSAR_MEM", "-Xms512m -Xmx512m -XX:MaxDirectMemorySize=1g")
+          .WithEnvironment("PULSAR_PREFIX_acknowledgmentAtBatchIndexLevelEnabled", "true")
+          .WithEnvironment("PULSAR_PREFIX_nettyMaxFrameSizeBytes", "5253120")
+          .WithEnvironment("PULSAR_PREFIX_transactionCoordinatorEnabled", "true")
+          .WithEnvironment("PULSAR_PREFIX_brokerDeleteInactiveTopicsEnabled", "true")
+          .WithEnvironment("PULSAR_STANDALONE_USE_ZOOKEEPER", "1")
+          .WithEnvironment("PULSAR_PREFIX_exposingBrokerEntryMetadataToClientEnabled", "true")
+          .WithEnvironment("PULSAR_PREFIX_brokerEntryMetadataInterceptors", "org.apache.pulsar.common.intercept.AppendBrokerTimestampMetadataInterceptor,org.apache.pulsar.common.intercept.AppendIndexMetadataInterceptor")
+
+          .WithEntrypoint("/bin/sh", "-c")
+          .WithCommand("while [ ! -f " + StartupScriptFilePath + " ]; do sleep 0.1; done; " + StartupScriptFilePath)
+
+          .WithWaitStrategy(Wait.ForUnixContainer())
```

Comment:
I think you need to add a proper wait strategy. Your wait strategy just waits until the container is running, but not until the service inside it is ready.

Reply:
Kafka waits for a log message, not just until the container is running:

https://github.com/testcontainers/testcontainers-dotnet/blob/ebedc9a5d23566c8353cff18c8320ffd2b6b9b58/src/Testcontainers.Kafka/KafkaBuilder.cs#L67